packages/manager/modules/pci/src/projects/project/notebooks/add/add.controller.js

```diff
@@ -0,0 +1,251 @@
+      (accumulator, label) => ({ ...accumulator, [label.key]: label.value }),
+      {},
+    );
+  }
+
+  static buildResourceBody(flavor, nbResources) {
+    const { resourcesPerUnit, gpuInformation } = flavor;
+    const { cpu, ...restOfResources } = resourcesPerUnit;
+
+    return {
+      [flavor.type]: nbResources,
+      flavor: flavor.id,
+      ...restOfResources,
+      ...gpuInformation,
+    };
+  }
+
+  static buildVolumesBody(volumes, region) {
+    return volumes.map((volume) => ({
+      container: volume.container,
+      region: region.name,
+      mountPath: volume.mountPath,
+      permission: volume.permission,
+      cache: false,
+    }));
+  }
+
+  static convertNotebookModel(notebookModel) {
+    const { name, nbResources } = notebookModel;
+    const { labels, volumes, selected } = notebookModel;
+    const { editor, framework, region, resource, privacy } = selected;
+
+    return {
+      env: {
+        editorId: editor.id,
+        frameworkId: framework.model.id,
+        frameworkVersion: framework.version.version,
+      },
+      labels: NotebookAddController.convertLabels(labels),
+      name,
+      region: region.name,
+      resources: NotebookAddController.buildResourceBody(
+        resource.flavor,
+        nbResources,
+      ),
+      volumes: NotebookAddController.buildVolumesBody(volumes, region),
+      unsecureHttp: NOTEBOOK_PRIVACY_SETTINGS.PUBLIC === privacy,
+    };
+  }
+
+  $onInit() {
+    this.messageContainer = 'pci.projects.project.notebooks.add';
+    this.loadMessages();
+    this.currentStep = 0;
+
+    // Notebook model
+    this.notebookModel = {
+      name: '',
+      labels: [],
+      repositoryURL: '',
+      mountPath: '',
+      nbResources: NOTEBOOK_RESOURCES.NB_RESOURCES,
+      volumes: [],
+      selected: {
+        editor: null,
+        framework: {
+          version: null,
+          model: null,
+        },
+        privacy: NOTEBOOK_PRIVACY_SETTINGS.RESTRICTED,
+        region: null,
+        useCase: null,
+        resource: {
+          usage: NOTEBOOK_RESOURCES.STANDARD,
+          flavor: null,
+          flavorType: NOTEBOOK_RESOURCES.STANDARD_FLAVOR,
+        },
+      },
+    };
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe(this.messageContainer);
+    this.messageHandler = this.CucCloudMessage.subscribe(
+      this.messageContainer,
+      { onMessage: () => this.refreshMessages() },
+    );
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  getConfigHeader(currentStep) {
+    if (currentStep > 0) {
+      return this.$translate.instant(
+        'pci_notebook_add_configuration_title_selected',
+        {
+          notebook: this.notebookModel.name,
+          labelsLength: this.notebookModel.labels.length,
+        },
+      );
+    }
+
+    return this.$translate.instant('pci_notebook_add_configuration_title');
+  }
+
+  getEditorHeader(currentStep) {
+    if (currentStep > 1) {
```

Comment:
i don't think you need to test for currentStep each time you want to get a translation since the oui-step form won't be displayed anyway

Reply:
I changed the way that i used.